### PR TITLE
UF-224: New Layout Editor

### PR DESCRIPTION
**Pull request by ederign**
API changes from https://github.com/uberfire/uberfire/pull/413

@pefernan do you mind to take a look?

This is a squash from new_layout_editor branch : 

cf639a9f8cedf7ff30a5d6bbdf6048ab50d1779e
1c03579a59e453a38005df47e10e5d901ea51318
e8df0edc5ee77ef0a98c067b5f4da589771dcd47
bc58182d33a3465f9782effeca951af068906ca8
3dd7a4025272f1c99000cfd0899ba7162c749ac9
a1330576729780933e9f77924d5c563cd5d598ac

### CODE REVIEW

File: livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java [hunk 1]
```diff
@@ -183,13 +187,13 @@ public Modal getConfigurationModal( final ModalConfigurationContext ctx ) {
         return propertiesRenderer.getView().getPropertiesModal();
     }
 
-    @Override
-    public void onDropComponent() {
+    public void onDropComponent(@Observes ComponentDropEvent event) {
+        //TODO convert from ComponentDropEvent for FieldDroppedEvent
```
Comment (ederign) [inline livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java]: @pefernan we have to fix this! could you please point me to the right direction?

File: livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java [hunk 2]
```diff
@@ -183,13 +187,13 @@ public Modal getConfigurationModal( final ModalConfigurationContext ctx ) {
...
         fieldDroppedEvent.fire( new FieldDroppedEvent( formId, fieldId ) );
     }
 
-    @Override
-    public void onRemoveComponent() {
+    public void onRemoveComponent(@Observes ComponentRemovedEvent event) {
+        //TODO convert from ComponentRemovedEvent for FieldDroppedEvent
```
Comment (ederign) [inline livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java]: @pefernan we have to fix this! could you please point me to the right direction?

File: livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java (hunk 1 above)
Comment (pefernan) [inline livespark-form-modeler/livespark-form-modeler-editor/livespark-form-modeler-editor-client/src/main/java/org/livespark/formmodeler/editor/client/editor/rendering/EditorFieldLayoutComponent.java]: @ederign look at your email... long explanation there

**Comment by pefernan** [comment]
+1 (Thanks @ederign)

**Pull request merged**